### PR TITLE
Add raw_document query support to backend

### PR DESCRIPTION
**Pull request by fridgepoet**
**What this PR does / why we need it**:
This recreates the raw_document query processing that exists in the frontend in the backend. It redirects any queries with only raw_document to the backend. It's very similar to the recent [raw_data work](https://github.com/grafana/opensearch-datasource/pull/203).

The query building is based on its equivalent in [Elasticsearch](https://github.com/grafana/grafana/blob/b39c8aec6c36a9557b147f628496af591cf1583e/pkg/tsdb/elasticsearch/data_query.go#L358) and the original PR is [here](https://github.com/grafana/grafana/pull/59741).
The result processing is based on its equivalent in [Elasticsearch](https://github.com/grafana/grafana/blob/b39c8aec6c36a9557b147f628496af591cf1583e/pkg/tsdb/elasticsearch/response_parser.go#L210). The original PR is [here](https://github.com/grafana/grafana/pull/63932).

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/opensearch-datasource/issues/195

**Special notes for your reviewer**:
* As mentioned in the [Elasticsearch PR](https://github.com/grafana/grafana/pull/63932), the order of the keys in the JSON will not be the same between the previous frontend implementation and this backend PR. 
  > results from backend are alphabetically ordered - this is done automatically while JSON.Marshall. From [docs](https://pkg.go.dev/encoding/json#Marshal): ...The map keys are sorted. This shouldn't be an issue at all, as the order of fields in json is not used anywhere.

* Another difference is that the `timestamp` will be an array, not just the value, same as in the [Elasticsearch PR](https://github.com/grafana/grafana/pull/63932). 
  > In json field (only field), @timestamp is array with value when processed trough backend (compared to just value when processed trough frontend).

  Although we can match the same behavior as the frontend by supplying only the single value from the array, if this more direct result has been acceptable for Elasticsearch users, the code will be simpler.

### CODE REVIEW

File: pkg/opensearch/lucene_handler.go [hunk 1]
```diff
@@ -59,8 +59,8 @@ func (h *luceneHandler) processQuery(q *Query) error {
 	}
 
 	switch {
-	case q.Metrics[0].Type == rawDataType:
-		processRawDataQuery(q, b, defaultTimeField)
+	case q.Metrics[0].Type == rawDocumentType, q.Metrics[0].Type == rawDataType:
```
Comment (iwysiu) [inline pkg/opensearch/lucene_handler.go:62]: Nit: Maybe i'm misremembering the syntax, but could you do:
```
switch q.Metrics[0].Type {
case rawDocumentType, rawDataType:
```

File: pkg/opensearch/response_parser.go [hunk 2]
```diff
@@ -81,6 +81,8 @@ func (rp *responseParser) getTimeSeries(timeField string) (*backend.QueryDataRes
 		switch {
 		case target.Metrics[0].Type == rawDataType:
 			queryRes = processRawDataResponse(res, timeField, queryRes)
+		case target.Metrics[0].Type == rawDocumentType:
```
Comment (iwysiu) [inline pkg/opensearch/response_parser.go:84]: same as the above switch comment

File: pkg/opensearch/response_parser.go [hunk 3]
```diff
@@ -131,6 +133,60 @@ func processRawDataResponse(res *es.SearchResponse, timeField string, queryRes b
 	return queryRes
 }
 
+func processRawDocumentResponse(res *es.SearchResponse, timeField, refID string, queryRes backend.DataResponse) backend.DataResponse {
+	documents := make([]map[string]interface{}, len(res.Hits.Hits))
+	for hitIdx, hit := range res.Hits.Hits {
+		doc := map[string]interface{}{
+			"_id":    hit["_id"],
+			"_type":  hit["_type"],
+			"_index": hit["_index"],
```
Comment (iwysiu) [inline pkg/opensearch/response_parser.go:142]: Do  need the `sort` and `highlight` fields that Elasticsearch is copying over?

File: pkg/opensearch/response_parser.go [hunk 4]
```diff
@@ -131,6 +133,60 @@ func processRawDataResponse(res *es.SearchResponse, timeField string, queryRes b
...
+		}
+		value := json.RawMessage(bytes)
+		fieldVector[i] = &value
+	}
+
+	isFilterable := true
+	field := data.NewField(refID, nil, fieldVector)
+	field.Config = &data.FieldConfig{Filterable: &isFilterable}
+
+	frames := data.Frames{}
```
Comment (iwysiu) [inline pkg/opensearch/response_parser.go:182]: nit: is there a reason not to do this as `frames := data.Frames{data.NewFrame(refID, field)}`
 or even `queryRes.Frames = data.Frames{data.NewFrame(refID, field)}`

**Review by iwysiu — COMMENTED** [review]
Mostly just style nits again, with one question about the hit fields

File: pkg/opensearch/response_parser.go (hunk 3 above)
Comment (fridgepoet) [reply pkg/opensearch/response_parser.go:142]: no, these are not in OpenSearch

**Review by iwysiu — APPROVED** [review]

**Pull request merged**